.github/workflows/maven.yml

```diff
@@ -0,0 +1,205 @@
+          The passphrase protecting the provided GPG Private Key.
+
+    outputs:
+      version: 
+        description: The detected Maven Version of the project as configured in the top level pom.xml
+        value: ${{ jobs.build.outputs.version }}
+      is-snapshot:
+        description: Whether the Maven build was for a SNAPSHOT version.
+        value: ${{ jobs.build.outputs.version != '' && contains(jobs.build.outputs.version, 'SNAPSHOT') }}
+      
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  cache-dependencies:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: read
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v4
+
+      - name: Install Java and Maven
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ inputs.JAVA_VERSION }}
+          distribution: ${{ inputs.JDK }}
+          cache: maven
+          # We intentionally don't configure credentials for the cache dependencies step.  Any dependencies MUST
+          # be in the public artifact repository so fail fast if a project is still relying on private dependencies
+          # that have not been published.
+
+      - name: Cache Maven Dependencies
+        run: |
+          mvn dependency:go-offline --batch-mode
+
+  build:
+    needs: cache-dependencies
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    env:
+      # These have to be exported into the environment due to how the setup-java action configures the Maven
+      # settings.xml file to avoid directly embedding credentials into it.
+      MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
+      MAVEN_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
+    permissions:
+      contents: read
+    outputs:
+      version: ${{ steps.project.outputs.version }}
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v4
+      
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        if: ${{ inputs.USES_DOCKERHUB_IMAGES && matrix.os == 'ubuntu-latest' }}
+        with:
+          username: ${{ secrets.DOCKER_READ_USER }}
+          password: ${{ secrets.DOCKER_READ_PWD }}
+
+      - name: Install Java and Maven
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ inputs.JAVA_VERSION }}
+          distribution: ${{ inputs.JDK }}
+          cache: maven
+          server-id: ${{ inputs.MAVEN_REPOSITORY_ID }}
+          # Forced to use indirection here, the setup-java action configures the Maven settings.xml to reference
+          # the credentials via environment variables
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v6
```

Comment:
We're passing our company's package signing private key to "Crazy Max's" obfuscated code. Do we know what this code is actually doing with it? https://github.com/crazy-max/ghaction-import-gpg/blob/master/dist/index.js

Reply:
Is this reassuring or ignorance at scale? ;)
https://github.com/digitalocean/terraform-provider-digitalocean/blob/main/.github/workflows/release.yml#L30
https://github.com/hashicorp/boundary/blob/main/.github/workflows/enos-run.yml#L130
https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/.github/workflows/release.yml#L20
https://github.com/bitwarden/server/blob/main/.github/workflows/version-bump.yml#L59

Reply:
> We would be passing our company's package signing private key to "Crazy Max's" obfuscated/minified code. Do we know what this code is actually doing with it? https://github.com/crazy-max/ghaction-import-gpg/blob/master/dist/index.js

Crazy Max is a fairly prolific open source developer involved heavily in the GitHub Actions space, he's also the main contributor on a number of high profile actions we already use e.g. https://github.com/docker/build-push-action/graphs/contributors

The `import-gpg-keys` action repository also contains the original source code - https://github.com/crazy-max/ghaction-import-gpg/tree/master/src